### PR TITLE
[UT] increase ut coverage for probe/util.go

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
 Increase ut coverage for probe/util.go

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by bart0sh** [comment]
one nit, otherwise lgtm
/lgtm

File: pkg/probe/util_test.go
Comment (saschagrunert) [inline pkg/probe/util_test.go:134]: Can we run those in parallel?
```suggestion
			t.Parallel()
			got, err := ResolveContainerPort(tt.args.param, tt.args.container)
```

File: pkg/probe/util_test.go
Comment (saschagrunert) [inline pkg/probe/util_test.go:46]: Can we run the test in parallel?
```suggestion
func TestResolveContainerPort(t *testing.T) {
	t.Parallel()
```

**Review by saschagrunert — APPROVED** [review]

File: pkg/probe/util_test.go
Comment (odinuge) [inline pkg/probe/util_test.go:133]:
```suggestion
	for _, tt := range tests {
		tt := tt	
```

Another nit, I think you should copy the `tt` var here to avoid a race when accessing it from the subtests running i parallel. You can eg. see [here](https://github.com/kubernetes/kubernetes/commit/6d85f947bf86d12e4459d31029c41d6049d40c0e).

Comment (odinuge) [reply pkg/probe/util_test.go:133]: Seems like the `pull-kubernetes-verify` is complaining about the same @my-git9 😄 

```
# k8s.io/kubernetes/pkg/probe
pkg/probe/util_test.go:136:37: loop variable tt captured by func literal
pkg/probe/util_test.go:136:52: loop variable tt captured by func literal
pkg/probe/util_test.go:137:23: loop variable tt captured by func literal
pkg/probe/util_test.go:138:68: loop variable tt captured by func literal
pkg/probe/util_test.go:141:14: loop variable tt captured by func literal
pkg/probe/util_test.go:142:59: loop variable tt captured by func literal
make[1]: *** [Makefile:339: vet] Error 
```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/115708/pull-kubernetes-verify/1625149019476987904

File: pkg/probe/util_test.go
Comment (bart0sh) [reply pkg/probe/util_test.go:43]: I don't see how this is resolved. `TestFindPortByName` is not changed.

**Comment by pacoxu** [comment]
/hold 
#114397
```
pkg/probe/dialer_others.go:1: : # k8s.io/kubernetes/pkg/probe [k8s.io/kubernetes/pkg/probe.test]
pkg/probe/util_test.go:199:6: TestResolveContainerPort redeclared in this block
	pkg/probe/util_test.go:96:6: other declaration of TestResolveContainerPort (typecheck)
//go:build !windows
```

**Comment by pacoxu** [comment]
It seems that this PR includes #114397 of TestResolveContainerPort.

**Comment by SergeyKanzhelev** [comment]
/hold

this seems to conflict with https://github.com/kubernetes/kubernetes/pull/114397/files https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-verify-govet-levee/1633971549713207296 in tide

**Comment by SergeyKanzhelev** [comment]
let's unhold (rebase will be needed) once another PR merged. See more here: https://kubernetes.slack.com/archives/CCK68P2Q2/p1678424529251249

File: pkg/probe/util_test.go
Comment (pacoxu) [reply pkg/probe/util_test.go:96]: Dup test name with https://github.com/kubernetes/kubernetes/pull/115708/files#diff-4a118db26fa2866f683a2cdf0f863edf63c9ac1b197e0c641b5ca6a3a669730bL46. 
Line 199 below

**Pull request merged**
